File: src/conclusao.py

class Heroi:
    def __init__(self, nome, idade, tipo):
        self.nome = nome
        self.idade = idade
        self.tipo = tipo

    def atacar(self):
        ataques = {
            'mago': 'usou magia',
            'guerreiro': 'usou espada',
            'monge': 'usou artes marciais',
            'ninja': 'usou shuriken'
        }
        ataque = ataques.get(self.tipo, 'ataque desconhecido')
        print(f"O {self.tipo} atacou usando {ataque}")

# Coletando inputs do usuário
nome = input("Digite o nome do herói: ")
idade = input("Digite a idade do herói: ")
tipo = input("Digite o tipo do herói (mago, guerreiro, monge, ninja): ")

# Criando e usando um objeto Heroi
heroi = Heroi(nome, int(idade), tipo)
heroi.atacar()

File: src/conclusao.js

class Heroi {
    constructor(nome, idade, tipo) {
        this.nome = nome;
        this.idade = idade;
        this.tipo = tipo;
    }
    atacar() {
        const ataques = {
            'mago': 'usou magia',
            'guerreiro': 'usou espada',
            'monge': 'usou artes marciais',
            'ninja': 'usou shuriken'
        };
        const ataque = ataques[this.tipo] || 'ataque desconhecido';
        alert(`o ${this.tipo} atacou usando ${ataque}`);
    }
}
// Coletando inputs do usuário
const nome = prompt("Digite o nome do herói: ");
const idade = prompt("Digite a idade do herói: ");
const tipo = prompt("Digite o tipo do herói (mago, guerreiro, monge, ninja): ");
// Criando e usando um objeto Heroi
const heroi = new Heroi(nome, idade, tipo);
heroi.atacar();